src/compiler/types.ts

```diff
@@ -3133,13 +3135,24 @@ namespace ts {
         readonly kind: SyntaxKind.JSDocComment;
         readonly parent: HasJSDoc;
         readonly tags?: NodeArray<JSDocTag>;
-        readonly comment?: string;
+        readonly comment?: NodeArray<JSDocText | JSDocLink>;

```

Comment:
If we decided we were sufficiently bothered by the increased memory usage, could this be `string | NodeArray`?  Alternatively, could we have `JSDoc` and `JSDocWithLinks`?